test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceControllerSpec.scala

```diff
@@ -0,0 +1,1328 @@
+            routes.SupportingEvidenceController.checkYourAnswers()
+          )
+        }
+
+        "the user has answered no to the question before and has answered yes and is on a single disposable journey" in {
+
+          val answers = IncompleteSupportingEvidenceAnswers(
+            doYouWantToUploadSupportingEvidence = Some(false),
+            evidences                           = List.empty,
+            expiredEvidences                    = List.empty
+          )
+
+          val (session, journey, draftReturn) = sessionWithSingleDisposalState(Some(answers))
+
+          val updatedDraftReturn =
+            draftReturn.copy(supportingEvidenceAnswers =
+              Some(
+                sample[IncompleteSupportingEvidenceAnswers]
+                  .copy(
+                    doYouWantToUploadSupportingEvidence = Some(true),
+                    evidences                           = List.empty,
+                    expiredEvidences                    = List.empty
+                  )
+              )
+            )
+
+          val updatedSession: SessionData =
+            session.copy(journeyStatus = Some(journey.copy(draftReturn = updatedDraftReturn)))
+
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(session)
+            mockStoreDraftReturn(
+              updatedDraftReturn,
+              journey.subscribedDetails.cgtReference,
+              journey.agentReferenceNumber
+            )(Right(()))
+            mockStoreSession(updatedSession)(Right(()))
+          }
+
+          checkIsRedirect(
+            performAction("supporting-evidence.do-you-want-to-upload" -> "true"),
+            routes.SupportingEvidenceController.checkYourAnswers()
+          )
+        }
+
+      }
+    }
+  }
+
+  "handling requests to upload supporting evidence" must {
+
+    "show check your answers page" when {
+
+      "the number of uploads has reached maximum allowed" in {
+
+        def performAction(): Future[Result] = controller.uploadSupportingEvidence()(FakeRequest())
+
+        val supportingEvidence = sample[SupportingEvidence]
+
+        val answers = IncompleteSupportingEvidenceAnswers(
+          doYouWantToUploadSupportingEvidence = Some(true),
+          evidences                           = List.fill(2)(supportingEvidence),
+          expiredEvidences                    = List.empty
+        )
+
+        val (session, _, _) = sessionWithSingleDisposalState(Some(answers))
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(session)
+        }
+
+        checkIsRedirect(
+          performAction(),
+          routes.SupportingEvidenceController.checkYourAnswers()
+        )
+      }
+    }
+
+    "show technical error page" when {
+
+      "upscan initiate call fails" in {
+        def performAction(): Future[Result] = controller.uploadSupportingEvidence()(FakeRequest())
```

Comment:
you could move this up a level and reuse it when you have the same thing in the tests below. A similar thing happens throughout the spec